src/core/components/courses/SlotCreationModal.vue

```diff
@@ -4,11 +4,14 @@
         Ajouter un <span class="text-weight-bold">créneau</span>
       </template>
       <ni-select in-modal caption="Etape" :options="stepOptions" :value="newCourseSlot.step" required-field
-        @blur="validations.step.$touch" :error="validations.step.$error" @input="update($event, 'step')" />
+        @blur="validations.step.$touch" :error="validations.step.$error" @input="updateStep" />
       <ni-datetime-range caption="Dates et heures" :value="newCourseSlot.dates" required-field disable-end-date
         :error="validations.dates.$error" @blur="validations.dates.$touch" @input="update($event, 'dates')" />
-      <ni-search-address :value="newCourseSlot.address" @input="update($event, 'address')" :error-message="addressError"
-        @blur="validations.address.$touch" :error="validations.address.$error" in-modal last />
+      <ni-search-address v-if="showField.address" :value="newCourseSlot.address" @input="update($event, 'address')"
+        :error-message="addressError" @blur="validations.address.$touch" :error="validations.address.$error" in-modal
+        last />
```

Comment:
Que penses-tu de mettre directement ici `this.getType(this.newCourseSlot.step) === ON_SITE` ? (en modifiant this.getType pour que si step n'existe pas ca renvoie false. Je trouve que c'est plus facile de savoir quand s'affiche quel champ